### PR TITLE
Adjust count when skipping partitions

### PR DESCRIPTION
**Pull request by lava**
Adjust count of outstanding queries when skipping some.

Also contains

### :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

File: libvast/src/system/index.cpp [hunk 1]
```diff
@@ -600,6 +600,9 @@ void index_state::schedule_lookups() {
       auto it = pending_queries.queries().find(qid);
       if (it == pending_queries.queries().end()) {
         VAST_WARN("{} tried to access non-existant query {}", *self, qid);
```
Comment (dominiklohmann) [inline libvast/src/system/index.cpp:602]: Spelling: It's existent with an e.

File: libvast/src/system/index.cpp [hunk 2]
```diff
@@ -600,6 +600,9 @@ void index_state::schedule_lookups() {
...
+        *cnt -= 1;
+        if (*cnt == 0)
+          --running_partition_lookups;
```
Comment (dominiklohmann) [inline libvast/src/system/index.cpp:605]: This snippet is now duplicated between here and `handle_completion` directly below. Can we put it into a lambda to give it a name?

E.g.,

```cpp
auto finish_query = [&] {
  if (--*cnt > 0)
    return false;
  --running_partition_lookups;
  return true;
};
```

**Review by dominiklohmann — APPROVED** [review]
The logic of this checks looks correct to me from trying to follow the flow of the function.

File: libvast/src/system/index.cpp (hunk 2 above)
Comment (tobim) [reply libvast/src/system/index.cpp:605]: You need to be careful with that. We call `handle_completion` when we get the response from the partition. If we create this helper closure outside of it the reference to `cnt` won't be valid later on. You would have to capture `cnt` by value and `finish_query` also by value in `handle_completion`.

**Pull request merged**